### PR TITLE
PPTP-281: initial auth integration

### PR DESCRIPTION
**Pull request by tombart**

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+govukTag: GovukTag,
+govukButton: GovukButton,
+taskList: taskList
+)
+
+@(tasks: Seq[TaskListViewModel])(implicit request: Request[_], messages: Messages)
+
+@govukLayout(title = Title("landingPage.title")) {
+
+<div class="govuk-!-width-two-thirds">
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html:39]: Why do we need this `div` here?

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    @pageTitle(text = messages("landingPage.title"))
+</div>
+
+<div id="contents">
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html:43]: Similar do this `div`. 
We should use GDS Design System as much as possible.

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/partials/phaseBanner.scala.html [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 @(phase: String)(implicit request: Request[_], messages: Messages)
 
-@link = {<a href="todo">@messages("feedback.link")</a>}
+@link = {<a href="todo" class="govuk-link">@messages("feedback.link")</a>}
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/views/partials/phaseBanner.scala.html:23]: :+1:

**Review by PauloPortugal — CHANGES_REQUESTED** [review]
Looks really good. Main points are:

1. probably we should rename the landing page to registration, as it is more meaningful
2. try to use as much as possible GDS design system on the templates

File: conf/app.routes [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # microservice specific routes
 
 GET        /assets/*file            controllers.Assets.versioned(path="/public", file: Asset)
-GET        /start                   uk.gov.hmrc.plasticpackagingtax.registration.controllers.StartController.displayStartPage()
\ No newline at end of file
+GET        /start                   uk.gov.hmrc.plasticpackagingtax.registration.controllers.StartController.displayStartPage()
+GET        /landing                 uk.gov.hmrc.plasticpackagingtax.registration.controllers.ChoiceController.displayPage()
```
[the remark this answers is not in the document]
Comment (tombart) [reply conf/app.routes]: done

File: conf/application.conf [hunk 5]
```diff
@@ -106,3 +114,8 @@ assets {
   url = "http://localhost:9032/assets/"
 }
 
+urls {
+  login = "http://localhost:9949/auth-login-stub/gg-sign-in"
+  loginContinue = "http://localhost:8503/plastic-packaging-tax/landing"
```
[the remark this answers is not in the document]
Comment (tombart) [reply conf/application.conf:119]: done

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/LandingController.scala [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                        "landing.businessContactInfo",
+                        "businessContactInfo",
+                        Some(Todo)
+      ),
+      TaskListViewModel("landing.applicantContactDetails",
+                        "landing.applicantContactInfo",
+                        "applicantContactInfo",
+                        Some(Todo)
+      )
+    )
```
[the remark this answers is not in the document]
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/LandingController.scala:67]: refactored a bit to decouple controller from fe text

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/components/taskList.scala.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+@import uk.gov.hmrc.plasticpackagingtax.registration.views.model.TaskListViewModel
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import uk.gov.hmrc.plasticpackagingtax.registration.views.html.components._
+
+@this(
+link: link,
+govukTag: GovukTag
+)
+
+@(viewModel: Seq[TaskListViewModel])(implicit messages: Messages)
```
[the remark this answers is not in the document]
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/components/taskList.scala.html:26]: refactored

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/components/taskList.scala.html [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                @task.statusTag.map { tag =>
+                @govukTag(Tag(content = Text(messages(tag.textMessage)), classes = tag.cssClass))
+                }
+            </div>
+        </div>
+    </li>
+    }
+    <li>
+
+</ol>
```
[the remark this answers is not in the document]
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/components/taskList.scala.html:51]: as discussed, no task list available yet

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html (hunk 2 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html:43]: removed

File: app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html (hunk 1 above)
Comment (tombart) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/views/landing_page.scala.html]: removed

**Review by tombart — COMMENTED** [review]
refactored

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/RegistrationController.scala [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+  val taskStatuses: Map[TaskName, TaskStatus] = Map(
+    TaskName.OrganisationDetails     -> TaskStatus.Completed,
+    TaskName.PlasticPackagingDetails -> TaskStatus.InProgress,
+    TaskName.BusinessContactDetails  -> TaskStatus.NotStarted
+  )
+
+  def displayPage(): Action[AnyContent] =
+    authenticate { implicit request =>
+      Ok(registration_page(taskStatuses))
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/RegistrationController.scala:43]: :+1:

**Review by PauloPortugal — APPROVED** [review]

**Pull request merged**